attachments/simple_engine/mesh_component.h

```diff
@@ -0,0 +1,461 @@
+                .location = 9,
+                .binding = 1,
+                .format = vk::Format::eR32G32B32A32Sfloat,
+                .offset = normalBase + 1u * vec4Size
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 10,
+                .binding = 1,
+                .format = vk::Format::eR32G32B32A32Sfloat,
+                .offset = normalBase + 2u * vec4Size
+            }
+        };
+        return attributeDescriptions;
+    }
+};
+
+/**
+ * @brief Structure representing a vertex in a mesh.
+ */
+struct Vertex {
+    glm::vec3 position;
+    glm::vec3 normal;
+    glm::vec2 texCoord;
+    glm::vec4 tangent;
+
+    bool operator==(const Vertex& other) const {
+        return position == other.position &&
+               normal == other.normal &&
+               texCoord == other.texCoord &&
+               tangent == other.tangent;
+    }
+
+    static vk::VertexInputBindingDescription getBindingDescription() {
+        vk::VertexInputBindingDescription bindingDescription(
+            0,                                  // binding
+            sizeof(Vertex),                     // stride
+            vk::VertexInputRate::eVertex        // inputRate
+        );
+        return bindingDescription;
+    }
+
+    static std::array<vk::VertexInputAttributeDescription, 4> getAttributeDescriptions() {
+        std::array<vk::VertexInputAttributeDescription, 4> attributeDescriptions = {
+            vk::VertexInputAttributeDescription{
+                .location = 0,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32Sfloat,
+                .offset = offsetof(Vertex, position)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 1,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32Sfloat,
+                .offset = offsetof(Vertex, normal)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 2,
+                .binding = 0,
+                .format = vk::Format::eR32G32Sfloat,
+                .offset = offsetof(Vertex, texCoord)
+            },
+            vk::VertexInputAttributeDescription{
+                .location = 3,
+                .binding = 0,
+                .format = vk::Format::eR32G32B32A32Sfloat,
+                .offset = offsetof(Vertex, tangent)
+            }
+        };
+        return attributeDescriptions;
+    }
+};
+
+/**
+ * @brief Component that handles the mesh data for rendering.
+ */
+class MeshComponent : public Component {
+private:
+    std::vector<Vertex> vertices;
+    std::vector<uint32_t> indices;
+
+    // Cached local-space AABB
+    glm::vec3 localAABBMin{0.0f};
+    glm::vec3 localAABBMax{0.0f};
+    bool localAABBValid = false;
+
+    // All PBR texture paths for this mesh
+    std::string texturePath;           // Primary texture path (baseColor) - kept for backward compatibility
```

Comment:
I don't see what you mean with "backward compatibility" here.
This is new, isn't it? Compatibility to what?